src/Microsoft.HttpRepl/Preferences/IPreferencesProvider.cs

```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+
+namespace Microsoft.HttpRepl.Preferences
+{
+    public interface IPreferencesProvider
+    {
+        string PreferencesFilePath { get; }
+        IReadOnlyDictionary<string, string> GetDefaultPreferences();
```

Comment:
Do the defaults vary by implementation?

Reply:
We discussed offline that no, they don't. I've modified the implementation to move it back out to `HttpState` for now. I want to take a look at what we discussed about changing the purpose of the `IPreferences` interface - from just persistence to fully controlling the preference values. But after looking into it, that "third level" of defaults we talked about is kind of spread around between `HttpState`, `RequestOrResponseConfig`, `RequestConfig`, `ResponseConfig`, `JsonConfig`. So I think we should look at changing that in a separate PR after this one is in since this one has already refactored so much.
Thoughts?